ZenPacks/zenoss/OpenStackInfrastructure/modeler/plugins/zenoss/OpenStackInfrastructure.py

```diff
@@ -800,6 +924,139 @@ def process(self, device, results, unused):
                     status=floatingip.get('status', 'UNKNOWN'),
                     )))
 
+        # volumes
+        volumes = []
+        for volume in results['volumes']:
+            if not volume.get('id', None):
+                continue
+
+            attachment = volume.get('attachments', [])
+            instanceId = ''
+            if len(attachment) > 0:
+                instanceId = attachment[0].get('server_id', '')
+            data=dict(
+                id=prepId('volume-{0}'.format(volume['id'])),
+                title=volume.get('name', ''),
+                volumeId=volume['id'],  # 847424
+                avzone=volume.get('availability_zone', ''),
+                created_at=volume.get('created_at', '').replace('T', ' '),
+                sourceVolumeId=volume.get('source_volid', ''),
+                host=volume.get('os-vol-host-attr:host', ''),
+                size=volume.get('size', 0),
+                bootable=volume.get('bootable', 'FALSE').upper(),
+                status=volume.get('status', 'UNKNOWN').upper(),
+                set_tenant=prepId('tenant-{0}'.format(volume.get('os-vol-tenant-attr:tenant_id',''))),
```

Comment:
This leaves it with "tenant-" as the tenant if there is no os-vol-tenant-attr:tenant_id.   Not really a great default- it will result in errors in the zenhub log with no real way to fix them..  I would omit set_tenant if there's no tenant id.